lnd_test.go

```diff
@@ -13201,6 +13201,159 @@ func testChannelBackupUpdates(net *lntest.NetworkHarness, t *harnessTest) {
 	}
 }
 
+// testExportChannelBackup tests that we're able to properly export either a
+// targeted channel's backup, or export backups of all the currents open
+// channels.
+func testExportChannelBackup(net *lntest.NetworkHarness, t *harnessTest) {
+	ctxb := context.Background()
+
+	// First, we'll create our primary test node: Carol. We'll use Carol to
+	// open channels and also export backups that we'll examine throughout
+	// the test.
+	carol, err := net.NewNode("carol", nil)
+	if err != nil {
+		t.Fatalf("unable to create new node: %v", err)
+	}
+	defer shutdownAndAssert(net, t, carol)
+
+	// With Carol up, we'll now connect her to Alice, and open a channel
+	// between them.
+	ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+	if err := net.ConnectNodes(ctxt, carol, net.Alice); err != nil {
+		t.Fatalf("unable to connect carol to alice: %v", err)
+	}
+
+	// Next, we'll open two channels between Alice and Carol back to back.
+	var chanPoints []*lnrpc.ChannelPoint
+	numChans := 2
+	chanAmt := btcutil.Amount(1000000)
+	for i := 0; i < numChans; i++ {
+		ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+		chanPoint := openChannelAndAssert(
+			ctxt, t, net, net.Alice, carol,
+			lntest.OpenChannelParams{
+				Amt: chanAmt,
+			},
+		)
+
+		chanPoints = append(chanPoints, chanPoint)
+	}
+
+	// Now that the channels are open, we should be able to fetch the
+	// backups of each of the channels.
+	for _, chanPoint := range chanPoints {
+		ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+		chanBackup, err := carol.ExportChannelBackup(ctxt, chanPoint)
+		if err != nil {
+			t.Fatalf("unable to fetch backup for channel %v: %v",
+				chanPoint, err)
+		}
+
+		// The returned backup should be full populated. Since it's
+		// encrypted, we can't assert any more than that atm.
+		if len(chanBackup.ChanBackup) == 0 {
+			t.Fatalf("obtained empty backup for channel: %v", chanPoint)
+		}
+
+		// The specified chanPoint in the response should match our
+		// requested chanPoint.
+		if chanBackup.ChanPoint.String() != chanPoint.String() {
+			t.Fatalf("chanPoint mismatched: expected %v, got %v",
+				chanPoint.String(),
+				chanBackup.ChanPoint.String())
+		}
+	}
+
+	// Before we proceed, we'll make two utility methods we'll use below
+	// for our primary assertions.
+	assertNumSingleBackups := func(numSingles int) {
+		ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+		req := &lnrpc.ChanBackupExportRequest{}
+		chanSnapshot, err := carol.ExportAllChannelBackups(ctxt, req)
```

Comment:
Feels like this can be incorporated with the previous test case, to avoid the overhead of running a new test

Reply:
There's a single test here? Commented on an outdated diff perhaps?

Reply:
I meant that `testExportChannelBackup` and `testChannelBackupUpdates` could be combined into one. In the test you could just first check that you get the backup updates, then you can manually export the channel backup, and make sure everything adds up.

Reply:
Yeah, but also nice to have some integration tests not be so "mega"...